service-app/internal/auth/middleware_mock.go

```diff
@@ -0,0 +1,39 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/ministryofjustice/opg-scanning/config"
+	"github.com/ministryofjustice/opg-scanning/internal/aws"
+	"github.com/ministryofjustice/opg-scanning/internal/logger"
+	"github.com/ministryofjustice/opg-scanning/internal/mocks"
+	"github.com/ministryofjustice/opg-scanning/internal/util"
+	"github.com/stretchr/testify/mock"
+)
+
+func PrepareMocks(mockConfig *config.Config, logger *logger.Logger) (*mocks.MockHttpClient, *Middleware, *aws.MockAwsClient, *JWTTokenGenerator) {
+	// Hash the password as required
+	passwordHash := util.HashPassword("test")
+
+	// Initialize the mock AWS client
+	mockAwsClient := new(aws.MockAwsClient)
+	mockAwsClient.On("GetSsmValue", mock.Anything, "/local/local-credentials").Maybe().Return(fmt.Sprintf("opg_document_and_d@publicguardian.gsi.gov.uk:%s", passwordHash), nil)
```

Comment:
There's no need to mock `GetSsmValue` since it's not called directly, which also means you can get rid of `passwordHash` and `util.HashPassword`